platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ProblemDefinition.java

```diff
@@ -57,11 +60,6 @@ public interface Problem {
      */
     Severity getSeverity();
```

Comment:
I tried to document it in the javadoc on the class level: that's not the case, the label and the category should be a unique identifier. The rest of the fields are there because they have a 1-to-1 relationship with the definition.

🤔  Is there a use-case where we need to deal with the same problem definition with different severities? My current understanding that  problems are inherently buildtree scoped, ie. they are collected as part of a single build invocation. In that context I cannot imagine the same problem definition having different severities. At the same time, this is not the case if we (ever) need to deal with problems on a broader (global) scope.

Reply:
I could imagine linter warnings:
 - In production code, something would be an error
 - In a test suite, it would be a warning

Interestingly, I realized why I had this feeling: for me, the definition should be a _unique key_ for a problem.